src/track/serato/tags.cpp

```diff
@@ -248,24 +257,11 @@ QList<CueInfo> SeratoTags::getCueInfos() const {
 
     const QList<CueInfo> cuesMarkers = m_seratoMarkers.getCues();
     for (const CueInfo& cueInfo : cuesMarkers) {
-        std::optional<int> index = findIndexForCueInfo(cueInfo);
-        if (!index) {
-            continue;
-        }
-        if (cueInfo.getType() == mixxx::CueType::Loop &&
-                (!cueInfo.getEndPositionMillis() ||
-                        *cueInfo.getEndPositionMillis() == 0)) {
-            // These entries are likely added via issue #11283
-            qWarning() << "Discard loop cue" << index << "found in markers with length of 0";
+        if (!isCueInfoValid(cueInfo)) {
             continue;
         }
-        if (cueInfo.getType() == mixxx::CueType::HotCue &&
-                (!cueInfo.getStartPositionMillis() ||
-                        *cueInfo.getStartPositionMillis() == 0) &&
-                (!cueInfo.getColor() ||
-                        *cueInfo.getColor() == mixxx::RgbColor(0))) {
-            // These entries are likely added via issue #11283
-            qWarning() << "Discard black hot cue" << index << "found in markers at position 0";
+        std::optional<int> pIndex = findIndexForCueInfo(cueInfo);
+        if (!pIndex) {
```

Comment:
The variable is not the `index` itself the name `pIndex` makes this visible. 
Yes, it is not a pointer, but thanks to `operator*()` and `operator->()` it can be used exactly like a pointer or smart pointer. 
From that point of view, it is a pointer to the value in itself. 
In this case the p prefix is a nice well known reminder for "don't forget to deference it" and "caution: it can be null" 
I have stumbled over during development.

Reply:
I still don't think we should use that prefix for anything but a pointer (actually we should prefix any variable simply based on its type or other attributes but thats outside the scope of the discussion). IMO the only legitimate reason to prefix pointers with `p` is because they have pointer semantics which make it much more difficult to reason about. `std::optional` has value semantics and is thus much easier to deal with. Adding `p` to something that has nothing to do with pointers will only cause confusing (and the next person looking at this code will likely change it back to the prefix-less version anyways with the same reason as before that its not a pointer). 

> In this case the p prefix is a nice well known reminder for "don't forget to deference it" and "caution: it can be null"
>  I have stumbled over during development.

How did you stumble over this? It should be nothing more than a compile error.

Reply:
How about introduce o prefix for this issue? 
`oIndex` can than easy destingushed from `index` that can be used without dereference.
`index = *oIndex`

Reply:
Possible, but I'm not a fan of prefixes and I don't think we need more of them. They have little benefit. They're only additional (arbitrary) rules for new contributors and they only get in the way when refactoring since now a change in type also needs a change in name. "I don't know how to use this common type so I need to remind myself how to use it every time" is not a good argument for type-based prefixes (the same applies to visibility prefixes `m_` but that's out of scope again).

Reply:
> There already are inconsistent prefixes everywhere in the code, e.g. pList for a QList of pointers 

Where should this be?

Reply:
Looks like these are already gone. But I remember that I stumbled across various inconsistent and arbitrary usage of type prefixes, sometimes even misleading.

Neither the compiler nor linter care. Better avoid them and focus on naming to communicate intentions, not types.